x-pack/legacy/plugins/ml/common/util/job_utils.js

```diff
@@ -438,6 +439,31 @@ export function basicJobValidation(job, fields, limits, skipMmlChecks = false) {
   };
 }
 
+export function basicDatafeedValidation(datafeed) {
+  const messages = [];
+  const valid = true;
+
+  if (datafeed) {
+    // if (_.isEmpty(datafeed.query)) {
```

Comment:
Has this been left in for a reason?

Reply:
yes, datafeed validation is an outstanding issue.
https://github.com/elastic/kibana/projects/30#card-27550795